### PR TITLE
Prevent multiple transactions of the same signer

### PR DESCRIPTION
**Pull request by earlbread**
This patch prevents multiple transactions of the same signer from being stored in one block to prevent transaction replay.  #125

### CODE REVIEW

File: Libplanet/Blocks/Block.cs [hunk 1]
```diff
@@ -390,6 +390,17 @@ internal void Validate(DateTimeOffset currentTime)
                 );
             }
 
+            var hasTxsOfSameSigner = Transactions
+                .GroupBy(tx => tx.Signer)
+                .Any(s => s.Count() > 1);
+
+            if (hasTxsOfSameSigner)
+            {
+                throw new MultipleTxsOfSameSignerException(
```
Comment (longfin) [inline Libplanet/Blocks/Block.cs]: It would be better if there is signer info who violate policy.

File: CHANGES.md [hunk 2]
```diff
@@ -91,6 +91,8 @@ To be released.
     [[#228], [#234]]
  -  `Swarm.StartAsync()` now does not call `Swarm.StopAsync()` anymore,
     therefore `Swarm.StopAsync()` should be explicitly called.  [[#236]]
+ -  `Block<T>` became to store only one transaction per signer.
```
Comment (dahlia) [inline CHANGES.md]: Since `Block<T>` does nothing with persist storage, we'd better use an other verb instead of “store” here, e.g.:

> A signer became to allowed to have only one transaction at most in a `Block<T>`.

File: Libplanet/Blocks/MultipleTxsOfSameSignerException.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
+using System;
+
+namespace Libplanet.Blocks
+{
+    /// <summary>
+    /// The exception that is thrown when the namespace of
+    /// <see cref="Block{T}"/> has multiple transactions of the same signer.
+    /// </summary>
+    [Serializable]
+    public sealed class MultipleTxsOfSameSignerException : Exception
```
Comment (dahlia) [inline Libplanet/Blocks/MultipleTxsOfSameSignerException.cs]: This added class also should be in the changelog.

File: Libplanet/Blocks/MultipleTxsOfSameSignerException.cs [hunk 4]
```diff
@@ -0,0 +1,22 @@
+using System;
+
+namespace Libplanet.Blocks
+{
+    /// <summary>
+    /// The exception that is thrown when the namespace of
```
Comment (dahlia) [inline Libplanet/Blocks/MultipleTxsOfSameSignerException.cs]: What the term “namespace” does mean here?

File: Libplanet/Blocks/Block.cs [hunk 5]
```diff
@@ -390,6 +390,17 @@ internal void Validate(DateTimeOffset currentTime)
...
+                    "Only one transaction per signer " +
+                    "can be stored in the block.");
```
Comment (dahlia) [inline Libplanet/Blocks/Block.cs]: How about this: “A signer is allowed to have only one transaction at most in a block.”

File: Libplanet/Blocks/MultipleTxsOfSameSignerException.cs (hunk 3 above)
Comment (dahlia) [inline Libplanet/Blocks/MultipleTxsOfSameSignerException.cs]: Would there be a better name for this? A type name having a preposition like “of” seems quite awkward to me.

File: Libplanet/Blocks/MultipleTxsOfSameSignerException.cs (hunk 4 above)
Comment (earlbread) [reply Libplanet/Blocks/MultipleTxsOfSameSignerException.cs]: My mistake. Fixed.

File: Libplanet/Blocks/MultipleTxsOfSameSignerException.cs (hunk 3 above)
Comment (earlbread) [reply Libplanet/Blocks/MultipleTxsOfSameSignerException.cs]: I renamed it to `SingersHaveMoreThanOneTxsException`.

File: Libplanet/Blocks/Block.cs [hunk 6]
```diff
@@ -261,6 +261,10 @@ public byte[] ToBencodex(bool hash, bool transactionData)
         /// any <see cref="IAction"/> of <see cref="Transactions"/> tries
         /// to update the states of <see cref="Address"/>es not included
         /// in <see cref="Transaction{T}.UpdatedAddresses"/>.</exception>
+        /// <exception cref="SimultaneousTxsException">Thrown when a
+        /// <see cref="Transaction{T}.Signer"/> has more than one
+        /// <see cref="Transaction{T}"/>s in <see cref="Transactions"/>.
```
Comment (dahlia) [inline Libplanet/Blocks/Block.cs]: The last word seems misleading (it's not incorrect though).   How about rewording a simple term “block” here?

```suggestion
        /// <see cref="Transaction{T}"/>s in a block.
```

Comment (earlbread) [reply Libplanet/Blocks/Block.cs]: Fixed.
